app/handlers/job_tracker.py

```
@@ -0,0 +1,227 @@
+    Represents a claimed job and provides operations to update its state.
+
+    This class wraps a `QueueProcessingRegistry` record and exposes async methods
+    to manage status transitions and job step updates.
+
+    Instances are constructed only through `JobTrackerManager.try_claim`.
+    """
+    def __init__(
+            self, worker_id: str, queue_name: str,
+            tracked_claim:QueueProcessingRegistryResponseDTO,
+            initial_step: Optional[JobLevels] = None,
+            queue_processing_registry_store=TortoiseQueueProcessingRegistryStore()
```

Ruff (0.12.2)

Do not perform function call `TortoiseQueueProcessingRegistryStore` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)